packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -268,10 +332,23 @@ def get_balance(
         :param token_id: the token id
         :return: the balance in a dictionary
         """
-        instance = cls.get_instance(ledger_api, contract_address)
-        balance = instance.functions.balanceOf(agent_address, token_id).call()
-        result = {token_id: balance}
-        return {"balance": result}
+        if ledger_api.identifier == "ethereum":
+            instance = cls.get_instance(ledger_api, contract_address)
+            balance = instance.functions.balanceOf(agent_address, token_id).call()
+            result = {token_id: balance}
+            return {"balance": result}
+        elif ledger_api.identifier == "cosmos":
+            msg = {
+                "balance":
+                    {
+                        "address": str(agent_address),
+                        "id": str(token_id)
+                    }
+            }
+            res = ledger_api.try_execute_wasm_query(contract_address, msg)
+            return json.loads(res)
```

Comment:
what does `json.loads return here...we need proper typed results; otherwise we have to run the code to understand.